x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -239,6 +247,32 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
         public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
             throw new IllegalArgumentException("[semantic_text] fields do not support sorting, scripting or aggregating");
         }
+
+        public QueryBuilder semanticQuery(InferenceResults inferenceResults, float boost, String queryName) throws IOException {
```

Comment:
@jimczi @carlosdelest 

Question: Should this method always return `MatchNoneQueryBuilder` when no inference results have been indexed (and therefore the field's dynamic mappings are not set), regardless of how `allowUnmappedFields` is configured. I'm leaning towards yes because those field mappings are an implementation detail from the user's POV.

Reply:
I would say so - it's not an unmapped field, but one that has no docs. Sounds like the good option to me.

Reply:
Done! This also allows us to delay fixing this KNN query issue: https://github.com/elastic/elasticsearch/issues/106846